### PR TITLE
End 2 end tests

### PR DESCRIPTION
**Pull request by houli**
Paging @laurmurclar to get in here
Not ready yet also expect some of these to break with UI changes.
Really these should be as general as possible without relying too much on DOM structure or very specific values.

Also, the way docker contexts work I have to copy in the fixture files which kind of sucks in keeping things in sync but I don't expect too much variance in the structure of the PML from now on

### CODE REVIEW

**Comment by laurmurclar** [comment]
Awesome, ty!

File: circle.yml [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
 test:
   pre:
     - mkdir -p $CIRCLE_TEST_REPORTS/exunit
     - mkdir -p $CIRCLE_TEST_REPORTS/pytest
+    - mkdir -p $CIRCLE_TEST_REPORTS/cucumber
+
   override:
-    - docker run -t -e "MIX_ENV=test" -e "CI=true" -v $CIRCLE_TEST_REPORTS/exunit:/test-reports panacea mix test
-    - docker run -t -v $CIRCLE_TEST_REPORTS/pytest:/test-reports asclepius pytest --junitxml=/test-reports/test-junit-report.xml
+    - docker-compose -f docker-compose.e2e.yml -f docker-compose.e2e.ci.yml up -d
```
Comment (c-brenn) [inline circle.yml:25]: is there a reason for the two `-f ...` flags here?

**Comment by houli** [comment]
In terms of running locally for now.
Build local asclepius and panacea
```bash
docker build -t tomtoothfairies/panacea panacea
docker build -t tomtoothfairies/asclepius asclepius
```

Then running tests you're working on
```bash
docker build -t tomtoothfairies/athloi athloi && docker-compose -f docker-compose.e2e.yml run athloi
```

If you define new features, cucumber will spit out the ruby code needed to define the steps

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -1,17 +1,21 @@
-version: '2'
+version: '2.1'
```
Comment (c-brenn) [inline docker-compose.dev.yml]: some of the files use `2.0` and some use `2.1` - should they all now be `2.1`?

**Review by c-brenn — COMMENTED** [review]
Really nice - just left a few questions

File: circle.yml (hunk 1 above)
Comment (houli) [reply circle.yml:25]: Yeah so I want to give him the option to be able to run these tests locally, but also in CI I want overrides to have a volume to collect junit xml.

File: docker-compose.dev.yml (hunk 2 above)
Comment (houli) [reply docker-compose.dev.yml]: 2.1 isn't supported by the docker on Circle, I was thinking of moving to 2.1 locally to add healthchecks but I'll probably revert that here and try it in a different PR

File: circle.yml (hunk 1 above)
Comment (c-brenn) [reply circle.yml:25]: ah I see, so it's merging the two configs through some sorcery?

File: athloi/features/step_definitions/file_upload_steps.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@
 end
 
 Then(/^I should see the unidentified drugs panel$/) do
-  panel = find('#unidentified-drugs-panel')
+  panel = find('#unidentified-drugs-panel', wait: 5)
```
Comment (c-brenn) [inline athloi/features/step_definitions/file_upload_steps.rb]: might as well wait as long as panacea will wait - 15 seconds

**Review by c-brenn — APPROVED** [review]
🚢 ☘️

File: athloi/features/step_definitions/shared_steps.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Given(/^I am on the home page$/) do
+  visit '/'
+end
+
+When(/^I select "([^"]*)"$/) do |filename|
+  # Capybara won't let you interact with hidden elements
+  execute_script("document.getElementById('file-input').classList.remove('hidden')")
```
Comment (c-brenn) [inline athloi/features/step_definitions/shared_steps.rb:7]: probably want to hide this again after adding the file.

**Comment by 22a** [comment]
🏷  ⚙️ 🗼 ⛩

:shipit:

**Pull request merged**